test/basics.jl

```diff
@@ -0,0 +1,45 @@
+string_types = [ASCIIStr, LatinStr, UTF8Str, UCS2Str, UTF16Str, UniStr]
+
+@testset "constructors" begin
+    for strtype in string_types
```

Comment:
Types are supposed to be mixed case in Julia, and for variables like this, a common convention is to use `T` or `S` etc.
Could you change `strtype` to `T`?